ESOF-docs/report4.md

```diff
@@ -0,0 +1,110 @@
+3. [Test Statistics](#test-statistics)
+4. [Bugs] (#bugs)
+4. [Critical Analysis](#critical-analysis)
+
+
+
+## Introduction
+
+> Software testability is the degree to which a software artifact (i.e. a software system, software module, requirements- or design document) supports testing in a given test context. If the testability of the software artifact is high, then finding faults in the system (if it has any) by means of testing is easier.
+
+>  _Definition from [Wikipedia]_
+[Wikipedia]: https://en.wikipedia.org/wiki/Software_testability
+
+In this report we will analyse **JPacman Framework**. 
+
+**JPacman** is an open source project developed by DELF College at the United States of America with the goal of teaching students how to do Unitary Tests in a more efficient way.
+
+Like the name says, this project consists of a Pacman-like game for 1 person that is missing some features to make the game more efficient.
+
+Therefore, the purpose of this project is to test all the functionalities that are developed until now to check if they are working well and if they can be improved in some way. 
+
+
+
+## Testability 
+
+By talking with some contributtors to the **JPacman Project** and by reading some information about the project on **Google** we learned that **JPacman** turns the test driven approach very easy and that the language lends itself well to testing because first **JPacman** project is written in Java that is a language with an easy approach and because the project itself is very well organized in different packages that turn a lot more easy tofind the information we are looking for to test. 
+
+This distribution makes also very easy to issolate the different components of the project we want to test that by consequence makes more easy to analyse the information gathered after testing which makes more easy to raise the Test Coverage and to find new bugs. For example, when we want to give a certain behavior to a set of types, we create a trait and implement it for each type and the trait itself can be tested independently. 
+
+During the tests, the compiler performs the heavy lifting when it comes to common bugs, so you can focus on other things and the language is a good match with unitary tests which leads to easily composable code. It is very simple to maintain coverage and to deploying the service. 
+
+For JPacman the usual practice is to have tests for some specific action like create board,move player,capture food,collision,etc.
+
+One of the biggest concerns about the testability of JPacman is the lack of support for handeling Exceptions and some type of Collisions. 
+
+
+As with any project, the project itself plays a big role. That said, for unit testing JPacman´s existing test tools are more than adequate. Integration testing may be more difficult, depending on what you're integrating, but Eclipse supports all the tools to create custom tests that sometimes may be hard to write but that in the end work very well.

```

Comment:
I think you're not familiar with the concept of integration testing. You're not integrating anything with something else, you're testing the integration between the conponents in a system, as opposed to unit testing.